File: src/net/merayen/elastic/backend/nodes/Environment.java
package net.merayen.elastic.backend.nodes;

import net.merayen.elastic.backend.mix.Mixer;
import net.merayen.elastic.backend.mix.Synchronization;

/**
 * Common 
 */
public class Environment {
	public final Mixer mixer;
	public final Synchronization synchronization;

	public Environment(Mixer mixer, Synchronization synchronization) {
		this.mixer = mixer;
		this.synchronization = synchronization;
	}
}

File: src/main/kotlin/net/merayen/elastic/backend/logicnodes/Environment.kt
package net.merayen.elastic.backend.logicnodes

import net.merayen.elastic.backend.data.project.Project
import net.merayen.elastic.backend.mix.Mixer
import net.merayen.elastic.backend.mix.Synchronization
import net.merayen.elastic.backend.nodes.LogicEnvironment
import net.merayen.elastic.backend.queue.Queue

/**
 * Environment that uses the current
 */
class Environment(mixer: Mixer, synchronization: Synchronization, sample_rate: Int, buffer_size: Int, project: Project)
    : LogicEnvironment(mixer, synchronization, sample_rate, buffer_size, project) {

    /**
     * The queue that should be used for any long-running process in the backend
     */
    val queue = Queue(4)
}
